src/block/block_ckpt.c

```diff
@@ -218,6 +218,8 @@ __wt_block_checkpoint_unload(WT_SESSION_IMPL *session, WT_BLOCK *block, bool che
 void
 __wt_block_ckpt_destroy(WT_SESSION_IMPL *session, WT_BLOCK_CKPT *ci)
 {
+    /* Make sure to own the live lock when running this on the live checkpoint. */
+
```

Comment:
This function is called for both live checkpoints with the lock, and for other checkpoints without the lock, and I didn't see a quick way to check within the function whether the checkpoint is live, which is why I left the comment instead of adding an assert.

FWIW, this function is also called from several places in a loop without the lock, and I'm not familiar enough with the code to tell whether any of those invocations could be with the live snapshot.

There is actually one place in `__wt_block_close` where we call this function without the lock: We first destroy the spinlock and then we call this to destroy the checkpoint. So strictly speaking, we are not always calling this function on the live checkpoint with the lock acquired.

How do you recommend that we should handle this?

1. Leave the comment as it is.
2. Remove the comment, if it's too confusing.
3. Find a way to check if this is the live checkpoint, so that we can add the assert (and update `__wt_block_close` accordingly). But for this option I would need a hint or two :)

Reply:
Then I would say something like:
We should hold the live lock here when running on the live checkpoint. But there is
no easy way to determine if the checkpoint is live so we cannot assert the locking here.

Reply:
That's a good idea, let me do that.